Godeps/_workspace/src/github.com/cloudflare/cfssl/crypto/pkcs11key/pkcs11key.go

```diff
@@ -159,35 +165,38 @@ func (ps *PKCS11Key) Destroy() {
 }
 
 func (ps *PKCS11Key) openSession() (session pkcs11.SessionHandle, err error) {
-	// Find slot by description
-	slots, err := ps.module.GetSlotList(true)
+	// Check if there is a PCKS11 token with slots. It has side-effects that
+	// allow the rest of the code here to work.
+	_, err = ps.module.GetSlotList(true)
```

Comment:
This is wrong, for two reasons.
1. The side-effect you're relying on is [not standard](http://docs.oracle.com/cd/E19253-01/816-4863/chapter2-9/index.html).  So this risks breaking with some modules.
2. You're throwing away usable information.  The list returned by this method *is* the list of slot IDs.

So you should read through this list to verify that the slot you're looking for is available, and fail before you even attempt `GetTokenInfo`.